### PR TITLE
made grid with drag function

### PR DESCRIPTION
**Pull request by rahul72parihar**
javascript 
- Generates grid
- Draws with event listener
- Slider for changing grid size

### CODE REVIEW

**Comment by rahul72parihar** [comment]
This is not responsive for mobile.
drag enter event listener have some issues

File: index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+let n = 16;
```
Comment (balwant-jtc) [inline index.js]: Instead of `n` use some meaningful name such as `gridSize`

File: index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const Container = document.getElementById("container");
```
Comment (balwant-jtc) [inline index.js]: Use camelCase for naming variables and functions.
So, `Container` will be `container`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
```
Comment (balwant-jtc) [inline index.js]: Remove this new line

File: index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const Slider = document.getElementById("slider");
```
Comment (balwant-jtc) [inline index.js]: Slider > slider

File: index.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+const SliderInput = document.getElementById("sliderValue");
```
Comment (balwant-jtc) [inline index.js]: SliderInput > sliderInput

File: index.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+Slider.oninput = function () {
```
Comment (balwant-jtc) [inline index.js]: Move the event handler logic to separate function, such as:

```
slider.addEventListener("input", handleSliderChange);

function handleSliderChange() {
  sliderValue.textContent = this.value;
  gridSize = this.value;
}
```

File: index.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  SliderInput.textContent = this.value;
+  n = this.value;
+};
+
+const generateBtn = document.getElementById("generate");
```
Comment (balwant-jtc) [inline index.js:4]: Place all DOM getters at one place at top, like


```
// Get the necessary elements from the DOM
const container = document.getElementById("container");
const slider = document.getElementById("slider");
const sliderValue = document.getElementById("sliderValue");
const generateBtn = document.getElementById("generate");

```

File: index.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+generateBtn.onclick = renderGrid;
```
Comment (balwant-jtc) [inline index.js]: generateBtn.addEventListener("click", renderGrid);

File: index.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+function renderGrid() {
+  let table = ``;
```
Comment (balwant-jtc) [inline index.js]: There is a better way to do this manipulations called `fragment`, read about it here :https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment

File: index.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    for (let j = 0; j < n; j++) {
+      table += `<div class = "atom"></div>`;
+    }
+    table += `</div>`;
+  }
+  Container.innerHTML = table;
+}
+renderGrid();
+
+document.addEventListener("dragenter", (e) => {
```
Comment (balwant-jtc) [inline index.js]:
```
document.addEventListener("dragenter", handleCellHighlight);

// Function to handle cell highlighting on drag
function handleCellHighlight(e) {
  if (e.target.classList.contains("atom")) {
    e.target.classList.add("green");
  }
}

```

File: index.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  if (e.target.classList.contains("atom")) {
+    e.target.classList.add("green");
+    console.log(e.target.classList);
```
Comment (balwant-jtc) [inline index.js]: remove console log

File: styles.css [hunk 12]
```diff
@@ -0,0 +1,47 @@
+*,
+*::before,
+*::after {
+  padding: 0;
+  margin: 0;
+}
+body {
+  text-align: center;
+}
+#container {
```
Comment (balwant-jtc) [inline styles.css:12]: Add 1 new line after every css style object

**Review by balwant-jtc — CHANGES_REQUESTED** [review]

**Pull request merged**